pkg/iac/scanners/terraform/parser/parser_test.go

```diff
@@ -1769,6 +1786,281 @@ module "invalid" {
 	}
 
 	require.NotContains(t, buf.String(), "failed to download")
+
+	// Verify module parents are set correctly.
+	expectedParents := map[string]string{
+		".":                     "",
+		"modules/city":          ".",
+		"modules/city/brooklyn": "modules/city",
+		"modules/city/queens":   "modules/city",
+	}
+
+	for _, mod := range modules {
+		expected, exists := expectedParents[mod.ModulePath()]
+		require.Truef(t, exists, "module %s does not exist in assertion", mod.ModulePath())
+		if expected == "" {
+			require.Nil(t, mod.Parent())
+		} else {
+			require.Equal(t, expected, mod.Parent().ModulePath(), "parent of module %q", mod.ModulePath())
+		}
+	}
+}
+
+// TestModuleParents sets up a nested module structure and verifies the
+// parent-child relationships are correctly set.
+func TestModuleParents(t *testing.T) {
```

Comment:
The test is very exhaustive (which is great!) but I wonder if we could reduce some setup complexity by using test golden files?

Reply:
If you think that is easier, I can dump these to actual files. The directory structure is flat, so all module paths are simple. So the file structure itself does not lend well to reading the actual hierarchy.

```
	// ./
	// ├── main.tf
	// └── modules
	//     ├── na
	//     ├── germany
	//     ├── wilmington
        //     ...
```

Reply:
Can we create the structure on disk the way we expect it to be tested? Basically the following

```
	// north-america
	// ├── denmark (greenland)
	// ├── united-states
	// │   ├── wilmington (north-carolina)
	// │   ├── wilmington (delaware)
	// │   ├── springfield (illinois)
	// │   ├── springfield (idaho)
	// │   ├── springfield (massachusetts)
	// │   └── new-york-city
	// │       └── manhattan
	// └── canada
	//     └── springfield (ontario)
	//
	// europe
	// ├── germany
	// │   └── berlin
	// ├── denmark
	// └── russia
	//     └── moscow
	//
	// asia
	// └── russia
	//     └── moscow

```

Reply:
I would prefer readability with creating the structure on disk. It is more verbose but it helps reduce testing logic and it's upkeep. Documenting structure in comments can also get stale quickly, if and when logic gets changed.